### DISCUSSION TITLE
What is the proper way to mark that a test init function failed ?

### DISCUSSION THREAD

**Discussion by cassepipe** [Snaipe/Criterion#435] [Q&A] [answered]
Let's say I want to write a test that needs to `open` a file before testing anything. 
If the `open` return `-1` then the test makes no sense. Is there a way to skip the test in this case and tell that the failure came from the init function ?

**Comment by Snaipe** [comment, marked as answer]
You should be able to call [cr_skip](https://criterion.readthedocs.io/en/latest/assert.html#c.cr_skip) with an appropriate message.

**Reply by cassepipe** [reply]
Oh I had seen that but it was not clear to me that it was its purpose. Thanks !
